microsoft/onnxruntime
# [js/react_native] Create ONNX Runtime React Native pipeline

This PR adds onnxruntime-react-native pipeline with e2e testing. There are lots of changes in this PR, but it's mostly because an example directory name is changed to e2e. Here the list of important changes.

- README.md explains how to run Android and iOS e2e tests and make npm package
- Android react native bridge library doesn't require to specify the version explicitly in gradle.properties. It picks up ONNX Runtime aar package in android/libs directory if exists or it downloads a published version
- Use the same gradle version 6.8.3 as mobile pipeline
- Use espresso test framework for Android e2e test 
- Add both Android and iOS UI test to check if inference runs properly
- Remove direct dependency on bridge library from e2e tests. Instead, create and install onnxruntime-common and onnxruntime-react-native npm packages, then start to run tests
- Add mac-react-native-ci-pipeline.yml to build, test, and publish onnxruntime-react-native npm package
- android-java-api-aar.yml is changed to get a pool name as a parameter
- pack-npm-packages.ps1 is changed to publish a react native package and to be able to run at cross-platform

The latest result of a pipeline can be found at [here](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=578482&view=results).
